dd-sdk-android-gradle-plugin/src/test/kotlin/com/datadog/gradle/plugin/KotlinCompilerTest.kt

```diff
@@ -0,0 +1,340 @@
+package com.datadog.gradle.plugin
+
+import com.datadog.gradle.plugin.kcp.DatadogPluginRegistrar
+import com.tschuchort.compiletesting.KotlinCompilation
+import com.tschuchort.compiletesting.SourceFile
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import org.intellij.lang.annotations.Language
+import org.jetbrains.kotlin.compiler.plugin.ExperimentalCompilerApi
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.times
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import org.mockito.quality.Strictness
+import kotlin.reflect.full.declaredFunctions
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@OptIn(ExperimentalCompilerApi::class)
+class KotlinCompilerTest {
+
+    private val datadogModifierSourceFileContent = SourceFile.kotlin(
+        DD_MODIFIER_CLASS_FILE_NAME,
+        DD_MODIFIER_SOURCE_FILE_CONTENT
+    )
+
+    @Mock
+    private lateinit var mockDDModifierCallback: () -> Unit
```

Comment:
just for the naming consistency

```suggestion
    private lateinit var mockDatadogModifierCallback: () -> Unit
```